community-operators [R] mongodb-atlas-kubernetes (0.4.0)

antonlisovenko opened the pull request:
Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [ ] Does your operator have [nested directory structure](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#create-a-bundle)?
* [ ] Have you selected the Project *Community Operator Submissions* in your PR on the right-hand menu bar?
* [ ] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md)?
* [ ] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [ ] Have you tested your Operator with all Custom Resource Definitions?
* [ ] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [ ] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator)?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?
* [ ] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#operator-icon) set?

### Updates to existing Operators

* [ ] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator)?
* [ ] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`?
* [ ] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#bundle-format) defined in the `package.yaml` or `annotations.yaml` ?
* [ ] Have you tested an update to your Operator when deployed via OLM?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?

### Your submission should not

* [ ] Modify more than one operator
* [ ] Modify an Operator you don't own
* [ ] Rename an operator - please remove and add with a different name instead
* [ ] Submit operators to both `upstream-community-operators` and `community-operators` at once
* [ ] Modify any files outside the above mentioned folders
* [ ] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [ ] Description about the managed Application and where to find more information
2. [ ] Features and capabilities of your Operator and how to use it
3. [ ] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [ ] Human readable name and 1-liner description about your Operator
* [ ] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-required-fields.md#categories)<sup>1</sup>
* [ ] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [ ] Links to the maintainer, source code and documentation
* [ ] Example templates for all Custom Resource Definitions intended to be used
* [ ] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator-capability-level.svg)

antonlisovenko commented:
@mvalarh not sure I understand why this one is failing...
Have tested on local Kind cluster (launched from `community-operators/community-operators/mongodb-atlas-kubernetes/0.4.0`):

```
operator-sdk olm install
docker build -f bundle.Dockerfile -t antonlisovenko/test-bundle:v0.4.0 .
docker push antonlisovenko/test-bundle:v0.4.0
operator-sdk run bundle docker.io/antonlisovenko/test-bundle:v0.4.0
INFO[0007] Successfully created registry pod: docker-io-antonlisovenko-test-bundle-v0-4-0
INFO[0007] Created CatalogSource: mongodb-atlas-kubernetes-catalog
INFO[0007] OperatorGroup "operator-sdk-og" created
INFO[0007] Created Subscription: mongodb-atlas-kubernetes-v0-4-0-sub
INFO[0011] Approved InstallPlan install-gwrzs for the Subscription: mongodb-atlas-kubernetes-v0-4-0-sub
INFO[0011] Waiting for ClusterServiceVersion "default/mongodb-atlas-kubernetes.v0.4.0" to reach 'Succeeded' phase
INFO[0011]   Waiting for ClusterServiceVersion "default/mongodb-atlas-kubernetes.v0.4.0" to appear
INFO[0023]   Found ClusterServiceVersion "default/mongodb-atlas-kubernetes.v0.4.0" phase: Pending
INFO[0024]   Found ClusterServiceVersion "default/mongodb-atlas-kubernetes.v0.4.0" phase: Installing
INFO[0053]   Found ClusterServiceVersion "default/mongodb-atlas-kubernetes.v0.4.0" phase: Succeeded
INFO[0053] OLM has successfully installed "mongodb-atlas-kubernetes.v0.4.0"

# deployment is up
k get deploy
NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
mongodb-atlas-operator   1/1     1            1           4m24s
```

I failed to find operator logs in the CI job...
Would appreciate any help!

mvalarh commented:
@J0zi Can you help?

J0zi commented:
I see your new push, tests are running but response to previous one:
@antonlisovenko, please click `Details` and `Show all hidden lines` afterwards.

We are failing on `CatalogSourcesUnhealthy` despite `READY`. You casn run our `kiwi` test [locally](https://github.com/operator-framework/community-operators/blob/master/docs/operator-test-suite.md), it is kubernetes, but I suppose it will way on the same issue.

Some this issue occurs when there is dependency on another operator.

antonlisovenko commented:
hi @J0zi 

Thanks for your recommendation, I tried to call the following:
```
bash <(curl -sL https://cutt.ly/WhkV76k) \
all \
community-operators/mongodb-atlas-kubernetes/0.4.0
Info: No labels defined
debug=0
Using ansible 2.10.6 on host ...

One can do 'tail -f /tmp/op-test/log.out' from second console to see full logs

Checking for kind binary ...
 [ Preparing testing container 'op-test' from 'quay.io/operator_testing/operator-test-playbooks:latest' ]
Test 'kiwi' for 'community-operators mongodb-atlas-kubernetes 0.4.0' ...
[kiwi] Reseting kind cluster ...

Failed with rc=1 !!!
Logs are in '/tmp/op-test/log.out'
```

The `log.out` shows the following:

```
Starting Ansible Pull at 2021-03-22 16:11:47
/usr/local/bin/ansible-pull -U https://github.com/operator-framework/operator-test-playbooks -C master -i localhost, -e ansible_connection=local -e run_upstream=true -e run_remove_catalog_repo=false upstream/local.yml -e run_prepare_catalog_repo_upstream=false --tags reset
[WARNING]: Could not match supplied host pattern, ignoring: Antons-MacBook-
Pro.local
[WARNING]: Could not match supplied host pattern, ignoring: Antons-MacBook-Pro
localhost | SUCCESS => {
    "after": "2f0d877c85788879a5094a79058a272f26da986a",
    "before": "2f0d877c85788879a5094a79058a272f26da986a",
    "changed": false,
    "remote_url_changed": false
}
[WARNING]: Could not match supplied host pattern, ignoring: Antons-MacBook-
Pro.local
[WARNING]: Could not match supplied host pattern, ignoring: Antons-MacBook-Pro

PLAY [Test operator main playbook] *********************************************

TASK [setup] *******************************************************************
[WARNING]:ok: [local Platform host]
darwin on host localhost is using the discovered Python
interpreter at /usr/bin/python, but future installation of another Python
interpreter could change the meaning of that path. See https://docs.ansible.com
/ansible/2.10/reference_appendices/interpreter_discovery.html for more
information.

TASK [Setting basic task variables] ********************************************
ok: [localhost]

TASK [Sets variables for docker] ***********************************************
ok: [localhost]

TASK [Sets variables for podman] ***********************************************
skipping: [localhost]

TASK [Set safety variables for production] *************************************
skipping: [localhost]

TASK [Sets variables for openshift] ********************************************
skipping: [localhost]
[WARNING]: reset_connection task does not support when conditional
[WARNING]: Reset is not implemented for this connection

TASK [install_kind : Check that the kind executable exists] ********************
ok: [localhost]

TASK [install_kind : Install kind binary] **************************************
skipping: [localhost]

TASK [Reset Kind with registry] ************************************************

TASK [reset_kind : include_tasks] **********************************************
included: /Users/alisovenko/.ansible/pull/Antons-MacBook-Pro.local/upstream/roles/reset_kind/tasks/reset.yml for localhost

TASK [reset_kind : Set the retry count] ****************************************
ok: [localhost]

TASK [reset_kind : Check that the kind executable exists] **********************
ok: [localhost]

TASK [reset_kind : Delete kind cluster] ****************************************
changed: [localhost]

TASK [reset_kind : Destroy current kind registry] ******************************
changed: [localhost]

TASK [reset_kind : Generating file for kind and registry] **********************
ok: [localhost]

TASK [reset_kind : Configuring Kind with registry] *****************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "/tmp/kind-with-registry.sh", "delta": "0:00:00.741524", "end": "2021-03-22 16:11:58.933485", "msg": "non-zero return code", "rc": 1, "start": "2021-03-22 16:11:58.191961", "stderr": "Error: No such container: kind-registry\nsed: 1: \"/etc/hosts\": extra characters at the end of h command", "stderr_lines": ["Error: No such container: kind-registry", "sed: 1: \"/etc/hosts\": extra characters at the end of h command"], "stdout": "292171b7541137bbb446ce51b3b1ef22f85f3e7aa276b04f212b1379a206f5fb", "stdout_lines": ["292171b7541137bbb446ce51b3b1ef22f85f3e7aa276b04f212b1379a206f5fb"]}

TASK [reset_kind : fail] *******************************************************
skipping: [localhost]

TASK [reset_kind : debug] ******************************************************
ok: [localhost] => {
    "msg": "Failed to install kind and registry - Retrying (2/5) ..."
}
```

Not sure I know what to do next... I've read the output in the job and don't see the reasons why the CatalogSource is not working - the Atlas Operator doesn't seem to have any dependencies on any other operators...

mvalarh commented:
@antonlisovenko do you have sudo without password set?

mvalarh commented:
@antonlisovenko you can also use our `Github action` in your project. More info [here](https://operator-framework.github.io/community-operators/action/)

J0zi commented:
/test all

antonlisovenko commented:
hi @J0zi  and @mvalarh 

I see all the operators in `community-operators` folder have `CustomResourceDefinition` of version `apiextensions.k8s.io/v1beta1` but our Operator uses `v1` instead - could it be the reason for the problem?

antonlisovenko commented:
have removed the `securityContext` in the Deployment as was recommended by Mikhail Mikhailitchenko

antonlisovenko commented:
I’m going to do some local testing, and this should be ready for another review after that - will add a comment

antonlisovenko commented:
hi @J0zi / @mvalarh 

thanks for your help, I've made local tests and the Operator looks good to me so the PR can be merged whenever you have time
